src/main/java/de/rototor/pdfbox/graphics2d/PdfBoxGraphics2DPaintApplier.java

```diff
@@ -248,6 +252,15 @@ private void applyPatternPaint(Paint paint, PaintApplierState state) throws IOEx
         state.contentStream.setStrokingColor(patternColor);
     }
 
+    /*
+     * Apache PDFBox Tiling Paint
+     */
+    private void applyTilingPaint(Paint paint, PaintApplierState state) throws IOException
+    {
+        TexturePaint texturePaint = getPrivateFieldValue(paint, "paint");
+        applyTexturePaint(texturePaint, state);
```

Comment:
Casting this to TexturePaint is likely calling for trouble. Why did you not simply call applyPaint() here with this paint?
Also you ignore the patternMatrix.

Reply:
Good point - that is a better way to handle it. Also added some exception handling to catch any issues with private field access.

I _think_ the patternMatrix should be accounted for already. In the TilingPaint class, the TexturePaint field is initialised from getImage(...) which applies the anchorRect and patternMatrix. The source and converted PDFs look the same.